---
# Documentation: https://sourcethemes.com/academic/docs/managing-content/

title: "The origins and genetic interactions of KRAS mutations are allele- and tissue-specific"
authors: [Joshua H. Cook, Giorgio E. M. Melloni, Doga C. Gulhan, Peter J. Park, Kevin M. Haigis]
date: 2021-03-22T00:00:00-00:00
doi: "10.1038/s41467-021-22125-z"

# Schedule page publish date (NOT publication's date).
publishDate: 2021-03-22T08:10:35-04:00

# Publication type.
# Legend: 0 = Uncategorized; 1 = Conference paper; 2 = Journal article;
# 3 = Preprint / Working Paper; 4 = Report; 5 = Book; 6 = Book section;
# 7 = Thesis; 8 = Patent
publication_types: ["2"]

# Publication name and optional abbreviated publication name.
publication: "Nature Communications"
publication_short: "Nat. Comm."

abstract: "Mutational activation of *KRAS* promotes the initiation and progression of cancers, especially in the colorectum, pancreas, lung, and blood plasma, with varying prevalence of specific activating missense mutations. Although epidemiological studies connect specific alleles to clinical outcomes, the mechanisms underlying the distinct clinical characteristics of mutant *KRAS* alleles are unclear. Here, we analyze 13,492 samples from these four tumor types to examine allele- and tissue-specific genetic properties associated with oncogenic *KRAS* mutations. The prevalence of known mutagenic mechanisms partially explains the observed spectrum of *KRAS* activating mutations. However, there are substantial differences between the observed and predicted frequencies for many alleles, suggesting that biological selection underlies the tissue-specific frequencies of mutant alleles. Consistent with experimental studies that have identified distinct signaling properties associated with each mutant form of KRAS, our genetic analysis reveals that each *KRAS* allele is associated with a distinct tissue- specific comutation network. Moreover, we identify tissue-specific genetic dependencies associated with specific mutant *KRAS* alleles. Overall, this analysis demonstrates that the genetic interactions of oncogenic *KRAS* mutations are allele- and tissue-specific, underscoring the complexity that drives their clinical consequences."

# Summary. An optional shortened abstract.
summary: "The *KRAS* gene is often mutated at several hotspot codons in cancer, resulting in similar, yet distinct, functional impacts on the KRAS protein. Here, the authors examine the genetic interactions of the different *KRAS* mutations across multiple cancer types and discover that *KRAS* mutations have allele- and tissue-specific mutagenic origins, comutation patterns, and dependency interactions.
"

tags: ["KRAS", "tissue-specificity", "cancer", "allele-specific", "computational biology", "bioinformatics"]
categories: []
featured: true

# Custom links (optional).
#   Uncomment and edit lines below to show custom links.
links:
  - name: PubMed
    url: https://www.ncbi.nlm.nih.gov/pubmed/?term=33753749
    icon_pack: ai
    icon: pubmed
  - name: Nat. Comm.
    url: https://www.nature.com/articles/s41467-021-22125-z
    icon_pack: fas
    icon: newspaper

url_pdf: https://www.nature.com/articles/s41467-021-22125-z.pdf
url_code: https://github.com/Kevin-Haigis-Lab/kras-allele-genetic-interactions
url_dataset: https://doi.org/10.6084/m9.figshare.14115569
url_poster:
url_project:
url_slides:
url_source:
url_video:

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder. 
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: "Fig. 3 The comutation networks of oncogenic KRAS alleles."
  focal_point: "Top"
  preview_only: false

# Associated Projects (optional).
#   Associate this publication with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `internal-project` references `content/project/internal-project/index.md`.
#   Otherwise, set `projects: []`.
projects: []

# Slides (optional).
#   Associate this publication with Markdown slides.
#   Simply enter your slide deck's filename without extension.
#   E.g. `slides: "example"` references `content/slides/example/index.md`.
#   Otherwise, set `slides: ""`.
slides: ""
---

<script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div data-badge-details="right" data-badge-type="medium-donut" data-doi="10.1038/s41467-021-22125-z" data-hide-no-mentions="true" class="altmetric-embed"></div>
